contracts/proto/deploy/v1/deploy.proto

```diff
@@ -175,6 +175,28 @@ message Api {
     }
 }
 
+// Declare a new websocket
+message Websocket {
+    // Target for handling new client connections
+    WebsocketTarget connect_target = 1;
+    // Target for handling client disconnections
+    WebsocketTarget disconnect_target = 2;
+    // Target for handling all other message types
+    // TODO: This may become a default handler in future and we can capture and filter
```

Comment:
@davemooreuws because this is modelled of off a serverless implementation where an external service is responsible for brokering the websocket connections, I imagine this would be handled by these services:
https://d2908q01vomqb2.cloudfront.net/1b6453892473a467d07372d45eb05abc2031647a/2018/12/18/websockets-arch.png

Do you have an example of how this would be done with something like AWS API Gateway?

Reply:
I was thinking more from the implementation side, which the users or SDKs can handle. All good, I see your point. 👯